homeworks/03-data-structure/src/SimpleBinaryTree.test.ts

```diff
@@ -0,0 +1,131 @@
+import BinaryTree, { TreeNode, TraverseType } from './BinaryTree'
+import SimpleBinaryTree from './SimpleBinaryTree'
+import {N, sort} from './TestHelper'
+
+describe("Simple binary tree", () => {
+  /** Test tree:
+   *          0
+   *        _/ \_
+   *       /     \
+   *      1       7
+   *     / \     / \
+   *    2   6   8   9
+   *   / \         /
+   *  3   4       10
+   *       \
+   *        5
+   */
+  const root = N(0,
+      N(1,
+        N(2,
+          N(3),
+          N(4, undefined, N(5))),
+        N(6)),
+      N(7,
+        N(8),
+        N(9, N(10)))
+    )
+  const tree = new SimpleBinaryTree(root)
+
+  const SingleRootValue = 20
+  const singleTree = new SimpleBinaryTree(N(SingleRootValue))
+
+  test("getRoot() returns root", () => {
+    expect(tree.getRoot()).toBe(root)
+  })
+
+  test("setRoot() replaces root and returns this", () => {
+    given:
+    const testTree = new SimpleBinaryTree(root)
+    const newRoot = N(25)
+
+    when:
+    const result = testTree.setRoot(newRoot)
+
+    then:
```

Comment:
This is the wrong usage: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/label
I guess they should be comments.

Reply:
That's Spock-style tests https://spockframework.org/spock/docs/1.3/spock_primer.html#_blocks . Here they are more for readability and fun, as there is no AST transformation in background (and Spock there is).

Reply:
I would say it's much more confused than readable or useful. Haven't seen such a style anywhere. 😅

Reply:
At least, no in JavaScript or TypeScript world.